pkg/util/hostname.go

```diff
@@ -161,6 +176,7 @@ func GetHostname() (string, error) {
 			err = ValidHostname(instanceID)
 			if err == nil {
 				hostName = instanceID
+				provider = "aws"
```

Comment:
Should we save the error from `getEC2Hostname` ? Same if the instance ID is not a valid hostname

Reply:
done